src/bases/createEvented.ts

```diff
@@ -61,6 +61,26 @@ function handlesArraytoHandle(handles: Handle[]): Handle {
 		}
 	};
 }
+/**
+ * Internal function to check if a target string matches a glob string that contains wildcards.
+ * Note: Due to limited use cases in event type name, currently only `*` that matches 0 or more characters is supported.
+ *
+ * @param globString The glob string that contains wildcards pattern
+ * @param targetString The string under test
+ * @return boolean match result
+ */
+
+function isGlobMatch(globString: string, targetString: string): boolean {
+	if (!globString || !targetString) {
+		return false;
+	}
+	const regex = new RegExp(`^${ globString.replace(/\*/g, '.*') }$`);
+
+	if (regex.test(targetString)) {
```

Comment:
Do we want to this regex test every time, for every listener attached? Feels like we'd only want to if we detected the supported wildcard character, i.e. `*`.

Also on the on we could pre-compile the `RegExp` if the type contains a wildcard and load that from the listenerMap?